### PR TITLE
Use smaller and compressed varients of buttons and form components

### PR DESCRIPTION
**Pull request by AMoo-Miki**
### Description
Replace instances of `EuiButton` that don't have an explicit sizing attribute to `EuiSmallButton*`.
Replace instances of `Eui<form elements>` that don't have density attributes to `EuiCompressed<form elements>`.
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/QueryResults/QueryResultsBody.tsx [hunk 1]
```diff
@@ -9,8 +9,9 @@ import { SortableProperties } from '@elastic/eui/lib/services';
 // @ts-ignore
 import {
   Comparators,
+  EuiSmallButton,
   EuiButton,
```
Comment (derek-ho) [inline public/components/QueryResults/QueryResultsBody.tsx:13]: Should this dependency be removed?

**Review by derek-ho — APPROVED** [review]

**Comment by derek-ho** [comment]
Snapshots should be dupated

File: public/components/QueryResults/QueryResultsBody.tsx (hunk 1 above)
Comment (AMoo-Miki) [reply public/components/QueryResults/QueryResultsBody.tsx:13]: No. It is used.

**Comment by AMoo-Miki** [comment]
Why does this repository have workflows that are known to fail?
* 2 workflows fail because the plugin cannot be installed
* the link checker is not configured properly
* the code is riddled with lint errors

Maybe disable these workflows for good open source citizenry.

**Pull request merged**